sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidJoinQueryRel.java

```diff
@@ -395,4 +365,14 @@ private static boolean computeRightRequiresSubquery(final DruidRel<?> right)
 
     return Pair.of(rightPrefix, signatureBuilder.build());
   }
+
+  private static DruidRel<?> getSomeDruidChild(final RelNode child)
+  {
+    if (child instanceof DruidRel) {
+      return (DruidRel<?>) child;
+    } else {
+      final RelSubset subset = (RelSubset) child;
```

Comment:
My understanding of Calcite's planner tells me that the children will either be single rels or will be a subset of equivalent rels. So DruidRel and RelSubset are the two cases that can happen.

I hope I understand the planner correctly — I can't point to any specific code that guarantees what I am saying is true.

Reply:
when I wrote this, I was wondering whether bindable parameters would change the node here somehow. The calcite query tests are pretty comprehensive around different types of JOINs and nested queries, that I feel pretty confident to agree with your understanding. Any chance you can test this with parameterized sql.

Reply:
I will look at adding some tests for this in the same followup as https://github.com/apache/druid/pull/9648#discussion_r406458569.

Reply:
subset of equivalent rels -> Does this imply that the cost of each RelNode will be equal? I saw that computeLeftRequiresSubquery does returns different depending on which child we pick from the list of RelList. This then result in a very very different cost since a child that results in requiring subquery will have very high cost and a child that doesnt will have a much lower cost. 

More specifically, I saw two RelNode in the list. One RelNode has a filter = null and the other has "filter":{"type":"selector","dimension":"v","value":"xa","extractionFn":null}